### PR TITLE
feat(storybook): add `addon-docs` support

### PR DESCRIPTION
**Pull request by SimonFinney**
## Proposed changes

- Add `addon-docs` to support detailed guidance documentation
- Refactor `addon-theme` to support theming in the 'Docs' tab
- Add `addon-docs` support for stories written in Component Story Format (CSF)
- Add `addon-controls` (replacement for `addon-knobs`) for better control in 'Docs' tab
- Delegate the configuration for addons and panel based on the format stories are written in
- Remove `meta` helper for flexibility in Storybook story definitions

### CODE REVIEW

**Comment by jendowns** [comment]
Hey @SimonFinney just double-checking -- are you meaning to only add the Docs capability in this PR, not any specific Docs content?

**Comment by SimonFinney** [comment]
@jendowns Sorry, I had to focus on some other things, but this PR has been revised based on recent changes and is ready for review. It now supports 'Docs' content for the 2 stories written in CSF -

- https://deploy-preview-741--ibm-security.netlify.app/?path=/docs/components-timeindicator--default
- https://deploy-preview-741--ibm-security.netlify.app/?path=/docs/components-trendingcard--default

File: .storybook/addons/addon-theme/Theme/index.js [hunk 1]
```diff
@@ -1,43 +1,24 @@
...
-      this.toggle(theme);
-    }
-  };
+export default ({ children, theme }) => {
+  useEffect(() => {
+    const { classList } = document.querySelector('html');
 
-  componentWillUnmount = () => this.toggle(this.props.theme);
+    classList.remove(
+      [...classList].find(className => className.includes(namespace))
```
Comment (Deriohig) [inline .storybook/addons/addon-theme/Theme/index.js:17]: what happens here if there is no returned element?

**Review by Deriohig — APPROVED** [review]
👍

**Pull request merged**